docs/tasks/manage-gpus/scheduling-gpus.md

```diff
@@ -4,154 +4,210 @@ approvers:
+Here's an example:
 
 ```yaml
-kind: pod
 apiVersion: v1
+kind: Pod
 metadata:
-  annotations:
-    scheduler.alpha.kubernetes.io/affinity: >
-      {
-        "nodeAffinity": {
-          "requiredDuringSchedulingIgnoredDuringExecution": {
-            "nodeSelectorTerms": [
-              {
-                "matchExpressions": [
-                  {
-                    "key": "alpha.kubernetes.io/nvidia-gpu-name",
-                    "operator": "In",
-                    "values": ["Tesla K80", "Tesla P100"]
-                  }
-                ]
-              }
-            ]
-          }
-        }
-      }
+  name: cuda-vector-add
 spec:
+  restartPolicy: OnFailure
   containers:
-    -
-      name: gpu-container-1
+    - name: cuda-vector-add
+      # https://github.com/kubernetes/kubernetes/blob/v1.7.11/test/images/nvidia-cuda/Dockerfile
+      image: "k8s.gcr.io/cuda-vector-add:v0.1"
       resources:
         limits:
-          alpha.kubernetes.io/nvidia-gpu: 2
+          nvidia.com/gpu: 1 # requesting 1 GPU
+```
+
+### Deploying NVIDIA GPU device plugin
+
+There are currently two device plugin implementations for NVIDIA GPUs:
+
+#### Official NVIDIA GPU device plugin
+
+The [official NVIDIA GPU device plugin](https://github.com/NVIDIA/k8s-device-plugin)
+has the following requirements:
+- Kubernetes nodes have to be pre-installed with NVIDIA drivers.
+- Kubernetes nodes have to be pre-installed with [nvidia-docker 2.0](https://github.com/NVIDIA/nvidia-docker)
+- nvidia-container-runtime configured as the [default runtime](https://github.com/NVIDIA/nvidia-docker/wiki/Advanced-topics#default-runtime)
+  for docker instead of runc.
+- NVIDIA drivers ~= 361.93
+
+To deploy the NVIDIA device plugin once your cluster is running and above
+requirements are satisfied:
+
 ```
+# For Kubernetes v1.8
+kubectl create -f https://raw.githubusercontent.com/NVIDIA/k8s-device-plugin/v1.8/nvidia-device-plugin.yml
 
-This will ensure that the pod will be scheduled to a node that has a `Tesla K80` or a `Tesla P100` Nvidia GPU.
+# For Kubernetes v1.9
+kubectl create -f https://raw.githubusercontent.com/NVIDIA/k8s-device-plugin/v1.9/nvidia-device-plugin.yml
+```
 
-### Warning
+Report issues with this device plugin to [NVIDIA/k8s-device-plugin](https://github.com/NVIDIA/k8s-device-plugin)
 
-The API presented here **will change** in an upcoming release to better support GPUs, and hardware accelerators in general, in Kubernetes.
+#### NVIDIA GPU device plugin used by GKE/GCE
 
-## Access to CUDA libraries
+The [NVIDIA GPU device plugin used by GKE/GCE](https://github.com/GoogleCloudPlatform/container-engine-accelerators/tree/master/cmd/nvidia_gpu)
+doesn't require using nvidia-docker and should work with any CRI compatible container
+runtime. It's tested on [COS](https://cloud.google.com/container-optimized-os/)
+and has experimental code for Ubuntu from 1.9 onwards.
 
```

Comment:
Suggestion: Spell out CRI and rephrase. Maybe "work with any container runtime that is compatible the Kubernetes Container Runtime Interface.

We should change COS to Container-Optimized OS. Google's branding guidelines say not to use acronyms except GCP and Cloud ML.

Reply:
Done